specification/sql/resource-manager/Microsoft.Sql/preview/2019-06-01-preview/DataWarehouseWorkloadClassifiers.json

```diff
@@ -0,0 +1,484 @@
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DatabaseNameParameter"
+          },
+          {
+            "name": "workloadGroupName",
+            "in": "path",
+            "description": "The name of the workload group from which to receive the classifier from.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "workloadClassifierName",
+            "in": "path",
+            "description": "The name of the workload classifier to create/update.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "description": "The properties of the workload classifier.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/DataWarehouseWorkloadClassifier"
+            }
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully updated the workload classifier.",
+            "schema": {
+              "$ref": "#/definitions/DataWarehouseWorkloadClassifier"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 FeatureDisabledOnSelectedEdition - User attempted to use a feature which is disabled on current database edition.\n\n * 400 OperationNotAllowedOnPausedDatabase - Operation is not allowed on a paused database.\n\n * 400 InvalidWorkloadGroupRequest - The request for the workload group is invalid.\n\n * 400 InvalidWorkloadGroupParameters - One or more parameters for workload group creation are incorrect.\n\n * 404 ServerNotInSubscriptionResourceGroup - Specified server does not exist in the specified resource group and subscription.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 DatabaseDoesNotExist - User has specified a database name that does not exist on this server instance.\n\n * 404 ResourceNotFound - The specified resource was not found.\n\n * 404 ResourceNotFound - The requested resource was not found.\n\n * 409 ConflictingSystemOperationInProgress - A system maintenance operation is in progress on the database and further operations need to wait until it is completed.\n\n * 409 ConflictingDatabaseOperation - There is already some operation on the database and the current operation should wait till it is done.\n\n * 503 DatabaseUnavailable - The operation failed because the database is unavailable.\n\n * 503 TooManyRequests - Requests beyond max requests that can be processed by available resources.\n\n * 504 RequestTimeout - Service request exceeded the allowed timeout."
+          },
+          "202": {
```

Comment:
Resource creation shouldn't return 202:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#creating-or-updating-resources

Reply:
202 is allowed for resource creation, see #6 of:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/Addendum.md#202-accepted-and-location-headers

Reply:
I followed existing examples which have the similar pattern of 202 for creation [https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedDatabases.json](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/sql/resource-manager/Microsoft.Sql/preview/2018-06-01-preview/managedDatabases.json)

Reply:
I forgot about this caveat, its not normally used.